anchore_engine/db/entities/policy_engine.py

```diff
@@ -795,36 +795,35 @@ def key_tuple(self):
 
     def to_nvd_reference(self) -> NVDReference:
         """
-        Returns an NVDReference object from this nvd vulnerability. Function to be used by non-nvd vulnerabilities
+        Returns an NVDReference object from this nvd vulnerability. Function to be used by non-nvd vulnerabilities for
+        populating an nvd reference
         """
 
         return NVDReference(
             vulnerability_id=self.name,
             severity=self.severity,
             link=self.link,
-            cvss=[
-                CVSS(
-                    version="2.0",
-                    base_score=self.get_max_base_score_nvd(2),
-                    exploitability_score=-1.0,
-                    impact_score=-1.0,
-                )
-            ],
+            cvss=self.get_all_cvss(),
         )
 
     def get_all_cvss(self) -> List[CVSS]:
         """
-        Return an empty list for backwards compatibility. Translates to no vendor cvss scores for an nvd vulnerability.
-        nvd cvss scores are available from get_all_nvd_references
+        Returns a list of CVSS objects for this vulnerability
         """
-
-        return []
+        return [
+            CVSS(
+                version="2.0",
+                base_score=self.get_max_base_score_nvd(2),
+                exploitability_score=-1.0,
+                impact_score=-1.0,
+            )
+        ]
 
     def get_all_nvd_references(self):
         """
-        Return a list with a single nvd reference. Function to be used by nvd vulnerability
+        Compatibility method. Returns empty list since an nvd vuln doesn't have any nvd refrences
         """
-        return [self.to_nvd_reference()]
+        return []
```

Comment:
hopefully this API change won't break any legacy things

Reply:
@nightfurys Where does this get used at?

Reply:
@Vijay-P There are functional and integration tests in place to verify and they are all passing 

@dspalmer99 It's used by the legacy provider to build out the cvss scores
https://github.com/anchore/anchore-engine/blob/2ffe7cd1db6c8ea47aadf1baea32a04610688933/anchore_engine/services/policy_engine/engine/vulns/providers.py#L377